app/functional_tests_fixtures/__init__.py

```diff
@@ -0,0 +1,460 @@
+    )
+    _create_service_email_reply_to(service.id, f"{test_email_username}+{environment}-reply-to@{email_domain}", False)
+
+    _create_service_permissions(service.id)
+
+    _create_service_sms_senders(service.id, "07700900500", True, inbound_number_id)
+    _create_service_sms_senders(service.id, "func tests", False, None)
+
+    _create_service_inbound_api(service.id, service_admin_user.id)
+
+    functional_test_config = f"""
+
+export FUNCTIONAL_TESTS_API_HOST={current_app.config['API_HOST_NAME']}
+export FUNCTIONAL_TESTS_ADMIN_HOST={current_app.config['ADMIN_BASE_URL']}
+
+export ENVIRONMENT={current_app.config['NOTIFY_ENVIRONMENT']}
+
+export FUNCTIONAL_TEST_EMAIL={func_test_user.email_address}
+export FUNCTIONAL_TEST_PASSWORD={functional_test_password}
+export TEST_NUMBER=07700900001
+
+export NOTIFY_SERVICE_API_KEY='{function_tests_govuk_key_name}-{govuk_service_id}-{api_key_notify.secret}'
+
+export FUNCTIONAL_TESTS_SERVICE_EMAIL='{service_admin_user.email_address}'
+export FUNCTIONAL_TESTS_SERVICE_EMAIL_AUTH_ACCOUNT='{email_auth_user.email_address}'
+export FUNCTIONAL_TESTS_SERVICE_EMAIL_PASSWORD={functional_test_password}
+export FUNCTIONAL_TESTS_SERVICE_NUMBER=07700900501
+
+export FUNCTIONAL_TESTS_SERVICE_ID='{service.id}'
+export FUNCTIONAL_TESTS_SERVICE_NAME='{service.name}'
+export FUNCTIONAL_TESTS_ORGANISATION_ID='{org.id}'
+export FUNCTIONAL_TESTS_SERVICE_API_KEY='{function_tests_live_key_name}-{service.id}-{api_key_live_key.secret}'
+export FUNCTIONAL_TESTS_SERVICE_API_TEST_KEY='{function_tests_test_key_name}-{service.id}-{api_key_test_key.secret}'
+export FUNCTIONAL_TESTS_API_AUTH_SECRET='{current_app.config['INTERNAL_CLIENT_API_KEYS']['notify-functional-tests'][0]}'
+
+export FUNCTIONAL_TESTS_SERVICE_EMAIL_REPLY_TO='{test_email_username}+{environment}-reply-to@{email_domain}'
+export FUNCTIONAL_TESTS_SERVICE_INBOUND_NUMBER=07700900500
+
+export FUNCTIONAL_TEST_SMS_TEMPLATE_ID={template2_id}
+export FUNCTIONAL_TEST_EMAIL_TEMPLATE_ID={template1_id}
+export FUNCTIONAL_TEST_LETTER_TEMPLATE_ID={template3_id}
+
+export MMG_INBOUND_SMS_USERNAME={current_app.config['MMG_INBOUND_SMS_USERNAME'][0]}
+export MMG_INBOUND_SMS_AUTH={current_app.config['MMG_INBOUND_SMS_AUTH'][0]}
+
+export REQUEST_BIN_API_TOKEN={request_bin_api_token}
+
+"""
+
+    if functional_test_env_file != "":
+        with open(functional_test_env_file, "w") as f:
+            f.write(functional_test_config)
+
+    if ssm_upload_path:
+        ssm = boto3.client("ssm")
+        response = ssm.put_parameter(
+            Name=ssm_upload_path,
+            Value=functional_test_config,
+            Type="SecureString",
+            Overwrite=True,
+        )
+
+        if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+            raise Exception("Failed to upload to SSM")
+
+
+def _create_user(name, email_address, password, auth_type="sms_auth", organisations=None, mobile_number=None):
```

Comment:
all the functions in this file are really similar to `tests/app/db.py`, which is reliably kept up to date with model changes since it's tested via unit tests.

can we reuse those functions? (it's probs ok if we need to move files around to avoid circular imports)

Reply:
As discussed these functions are idempotent vs the db.py ones. We also need to move the db.py out of tests and into the app. We decided it wasn't worth it in the end.